integration/chaincode/marbles_private/chaincode.go

```diff
@@ -377,6 +380,84 @@ func (t *MarblesPrivateChaincode) delete(stub shim.ChaincodeStubInterface, args
 	return shim.Success(nil)
 }
 
+// =====================================================
+// purge - remove a marble key/value pair from state and
+// remove all trace of private details
+// =====================================================
+func (t *MarblesPrivateChaincode) purge(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	fmt.Println("- start delete marble")
+
+	type marblePurgeTransientInput struct {
+		Name string `json:"name"`
+	}
+
+	if len(args) != 0 {
+		return shim.Error("Incorrect number of arguments. Private marble name must be passed in transient map.")
+	}
+
+	transMap, err := stub.GetTransient()
+	if err != nil {
+		return shim.Error("Error getting transient: " + err.Error())
+	}
+
+	marblePurgeJsonBytes, ok := transMap["marble_purge"]
+	if !ok {
+		return shim.Error("marble_purge must be a key in the transient map")
+	}
+
+	if len(marblePurgeJsonBytes) == 0 {
+		return shim.Error("marble_purge value in the transient map must be a non-empty JSON string")
+	}
+
+	var marblePurgeInput marblePurgeTransientInput
+	err = json.Unmarshal(marblePurgeJsonBytes, &marblePurgeInput)
+	if err != nil {
+		return shim.Error("Failed to decode JSON of: " + string(marblePurgeJsonBytes))
+	}
+
+	if len(marblePurgeInput.Name) == 0 {
+		return shim.Error("name field must be a non-empty string")
+	}
+
+	// to maintain the color~name index, we need to read the marble first and get its color
+	valAsbytes, err := stub.GetPrivateData("collectionMarbles", marblePurgeInput.Name) // get the marble from chaincode state
+	if err != nil {
+		return shim.Error("Failed to get state for " + marblePurgeInput.Name)
+	} else if valAsbytes == nil {
+		return shim.Error("Marble does not exist: " + marblePurgeInput.Name)
+	}
+
+	var marbleToDelete marble
+	err = json.Unmarshal([]byte(valAsbytes), &marbleToDelete)
+	if err != nil {
+		return shim.Error("Failed to decode JSON of: " + string(valAsbytes))
+	}
+
+	// delete the marble from state
+	err = stub.DelPrivateData("collectionMarbles", marblePurgeInput.Name)
+	if err != nil {
+		return shim.Error("Failed to delete state:" + err.Error())
+	}
+
+	// Also delete the marble from the color~name index
+	indexName := "color~name"
+	colorNameIndexKey, err := stub.CreateCompositeKey(indexName, []string{marbleToDelete.Color, marbleToDelete.Name})
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+	err = stub.DelPrivateData("collectionMarbles", colorNameIndexKey)
```

Comment:
stub.PurgePrivateData?

Reply:
I think I had intended to do a delete, followed by a purge, since that seems likely to be the most common scenario.

Reply:
Same as above